### PR TITLE
Task/cra 71

### PR DESCRIPTION
**Pull request by akanksha-einn**
1. Changed Code of Instructions to test Instruction API with updated API document.
2. Added unit test cases to test Instructions Api, to get Instruction with id from server.

### CODE REVIEW

File: android/MaterialDesignTest/app/src/main/java/com/testexample/materialdesigntest/ui/instructions/InstructionActivity.kt [hunk 1]
```diff
@@ -31,7 +31,7 @@ class InstructionActivity : AppCompatActivity(), InstructionsContract.View {
         val calender = Calendar.getInstance()
         val date = DateFormat.getDateInstance().format(calender.time)
 
-        presenter.fetchInstructions("802", "29-06-2020")
+        presenter.fetchInstructions("802")
```
Comment (nidhi-einn) [inline android/MaterialDesignTest/app/src/main/java/com/testexample/materialdesigntest/ui/instructions/InstructionActivity.kt:34]: What is "802"?

File: android/MaterialDesignTest/app/src/main/java/com/testexample/materialdesigntest/ui/instructions/InstructionPresenter.kt [hunk 2]
```diff
@@ -29,15 +29,20 @@ class InstructionPresenter(private var view: InstructionsContract.View?):
...
+                    {success ->
+                        view!!.showInstructions(success)
+
+                        Log.d(TAG, "Fetching Instruction From Remote")
                     },
-                    {error -> println(error.toString())},
-                    { println("instructions fetched successfully")}
+                    {err ->
+                        println(err.localizedMessage)
+                        Log.d(TAG, "Error in fetching Instruction from Remote")
```
Comment (nidhi-einn) [inline android/MaterialDesignTest/app/src/main/java/com/testexample/materialdesigntest/ui/instructions/InstructionPresenter.kt:43]: Why it is Log.d? if it is an error log then it should be like Log.e
TAG should be in small

File: android/MaterialDesignTest/app/src/test/java/com/testexample/materialdesigntest/data/network/repository/InstructionsRemoteRepoTest.kt [hunk 3]
```diff
@@ -31,10 +34,167 @@ class InstructionsRemoteRepoTest {
...
+                        { error -> println("callInstructionsApi_withInvalidTokenNId(): ${error.localizedMessage}")
+                            failure = error.message.toString()
+                        })
+
+        if (failure.isBlank()){
+            fail("callInstructionsApi_withInvalidTokenNId(): Verification failed as received Instructions")
+        }
+        else {
+            assertEquals("callInstructionsApi_withInvalidTokenNId(): Verification Failed as received $failure", "701 Access denied", failure)
+        }
```
Comment (nidhi-einn) [inline android/MaterialDesignTest/app/src/test/java/com/testexample/materialdesigntest/data/network/repository/InstructionsRemoteRepoTest.kt:198]: Please add logs in this file.

Comment (akanksha-einn) [reply android/MaterialDesignTest/app/src/test/java/com/testexample/materialdesigntest/data/network/repository/InstructionsRemoteRepoTest.kt:198]: Done

File: android/MaterialDesignTest/app/src/main/java/com/testexample/materialdesigntest/ui/instructions/InstructionPresenter.kt (hunk 2 above)
Comment (akanksha-einn) [reply android/MaterialDesignTest/app/src/main/java/com/testexample/materialdesigntest/ui/instructions/InstructionPresenter.kt]: Done,
 This TAG should be in Upper Case as per convention.

File: android/MaterialDesignTest/app/src/main/java/com/testexample/materialdesigntest/ui/instructions/InstructionActivity.kt (hunk 1 above)
Comment (akanksha-einn) [reply android/MaterialDesignTest/app/src/main/java/com/testexample/materialdesigntest/ui/instructions/InstructionActivity.kt:34]: This part of code has been changed by @shresthdeepgupta, he will commit his changes after merge of this branch to shresth_testing

**Pull request merged**